apps/src/aiTutor/redux/aiTutorRedux.ts

```diff
@@ -143,29 +136,32 @@ const aiTutorSlice = createSlice({
     setScriptId: (state, action: PayloadAction<number | undefined>) => {
       state.scriptId = action.payload;
     },
-    addChatMessage: (state, action: PayloadAction<ChatCompletionMessage>) => {
-      const newMessageId =
-        state.chatMessages[state.chatMessages.length - 1].id + 1;
-      const newMessage = {
-        ...action.payload,
-        id: newMessageId,
-      };
-      state.chatMessages.push(newMessage);
+    addChatMessage: (
+      state,
+      {payload}: PayloadAction<ChatCompletionMessage>
+    ) => {
+      state.chatMessages.push(payload);
     },
     clearChatMessages: state => {
       state.chatMessages = initialChatMessages;
     },
     setIsWaitingForChatResponse: (state, action: PayloadAction<boolean>) => {
       state.isWaitingForChatResponse = action.payload;
     },
-    updateChatMessageStatus: (
+    updateLastChatMessage: (
       state,
-      action: PayloadAction<{id: number; status: AITutorInteractionStatusValue}>
+      action: PayloadAction<Partial<ChatCompletionMessage>>
     ) => {
-      const {id, status} = action.payload;
-      const chatMessage = state.chatMessages.find(msg => msg.id === id);
-      if (chatMessage) {
-        chatMessage.status = status;
+      if (state.chatMessages.length > 0) {
+        const lastMessage = state.chatMessages[state.chatMessages.length - 1];
+        const payloadKeys = Object.keys(action.payload) as Array<
+          keyof ChatCompletionMessage
+        >;
+
+        payloadKeys.forEach(key => {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          (lastMessage as any)[key] = action.payload[key];
```

Comment:
I tried to make this a method that would update the most recent method with any keys that were contained in the payload, but TypeScript was not happy 😅 Opinions @molly-moen? I could split this out into two methods to update the status and/or id.

Reply:
what was the error you were getting?

Reply:
`` Type 'string | number | undefined' is not assignable to type 'never'.
  Type 'undefined' is not assignable to type 'never'.```

My guess is that TypeScript can't be sure that the type of `action.payload[key]` matches the type of `lastMessage[key]` because it's not making type inference when doing dynamic assignments like this.

Reply:
The `(lastMessage as any)` fixed it, but I assume we want to avoid that.

Reply:
gotcha yeah it can't track the key being part of the `ChatCompletionMessage` function. It's not ideal to do this because we want to avoid `any`, but it might be ok given we know the keys should be accurate. However, would something like this work?
```
state.chatMessages[state.chatMessages.length - 1] = {...lastMessage, ...action.payload}
```
(Alternatively you could pop/push `lastMessage` to avoid the wordiness with setting the value)

Reply:
That was a good call, thanks! I'll remember to merge/spread like this in the future to avoid weirdness with dynamic assignment.